vmm/src/vm_config.rs

```diff
@@ -54,6 +54,22 @@ pub fn default_cpuconfig_max_phys_bits() -> u8 {
     DEFAULT_MAX_PHYS_BITS
 }
 
+// Some runtime objects, such as net devices and memory zones, may
+// be provided to Cloud Hypervisor via file descriptors constructed
+// "externally". For example, when a unix domain socket is used to
+// configure/create a new VM via VmCreate HTTP API command, the API
+// client may send file descriptors via SCM_RIGHTS UDS side-channel.
+// `ExternalFdsConfig` will then contain the list of devices (`ids`)
+// that are to receive the FDs; in the API call `ExternalFdsConfig::fds`
+// field will initially be empty, and the API handler will populate
+// it with file descriptors received via the "side channel".
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct ExternalFdsConfig {
+    pub ids: Vec<String>,
```

Comment:
Please add a comment. `/// Component IDs (virtio-net devices, memory, ...) that will be associated with external FDs`